lock-api/src/test/java/com/palantir/lock/client/LeasedLockTokenTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import java.util.UUID;
+
+import org.junit.Test;
+
+import com.palantir.common.time.NanoTime;
+import com.palantir.lock.v2.LeaderTime;
+import com.palantir.lock.v2.LeadershipId;
+import com.palantir.lock.v2.Lease;
+import com.palantir.lock.v2.LockToken;
+
+public class LeasedLockTokenTest {
```

Comment:
think there's a fair bit of behaviour in LeasedLockToken which is not tested here; stuff like 'we shouldn't allow an update from a different user', 'we shouldn't allow it to be updated into the past', 'it can be valid past refresh', etc. This test class doesn't really test the contract of the leased lock token.

Reply:
there's stuff around the surrogate id not being the same as the server's lock token